### ISSUE TITLE
Working with Azure WebJobs with Hybrid SQL Connection

### ISSUE THREAD

**Issue by polymathic9999** [dyatchenko/ServiceBrokerListener#36]
I have been using SqlDependencyEx with a console App successfully. I get the notifications, the actual data changes etc.

I have to move this executable to an Azure WebJob, and need to work with an on-prem SQL server database for now. Would SqlDependencyEx still work in this scenario?

Awesome project, thanks.

Swanand

**Comment by dyatchenko** [comment]
Hi,

Thanks for using the project. If you use Azure SQL PaaS then it won't work because this version of SQL server is very restricted. And it doesn't have Service Broker which is used for notifications. Service Broker is the only solution to have some sort of notification in SQL server. Thus, none of the existing notification solutions will work there.
They may implemented it already, but I'm not sure. I checked it about a year ago and it didn't work. 

However, you can use an Azure IaaS VM and configure SQL server there. 

Thanks,
Dmitrii

**Comment by polymathic9999** [comment]
Thanks for your quick reply.

The database is a regular SQL server 2014 instance on premises, it is NOT
an Azure SQL database. As I mentioned, I used a console app on my local
machine and was able to receive notifications.

I am trying to move the console app (exe) to an Azure WebJob, so I have to
use a hybrid SQL connector to connect to an on-premise database.
Theoretically, it should work.

The reason we are not using a VM is cost. Maybe it will change if this is
the only way.

Thanks again.

Best regards
Swanand

**Comment by Siliconrob** [comment]
Your data is on premise, but your compute is off premise?  You do know that latency will kill you in this situation?  And compute should be as close to data as possible to avoid network issues and costs.

Natural evolution of your console app to something that runs all the time is a windows service.  Check out [https://github.com/Topshelf/Topshelf](TopShelf)

**Comment by dyatchenko** [comment]
If there are no updates I'm closing it.

**Comment by polymathic9999** [comment]
Thanks Dmitriy*.*

Sorry, your email was kind of lost in my inbox and I couldn't reply earlier.
I did exactly as you suggested, the windows service solutions works just
fine.

Had another question regarding permissions: I granted the following to the
user (used in my connection string).

GRANT CREATE PROCEDURE TO [servicebroker];
GRANT CREATE SERVICE TO [servicebroker];
GRANT CREATE QUEUE  TO [servicebroker];
GRANT REFERENCES ON CONTRACT::[DEFAULT] TO [servicebroker];
GRANT SUBSCRIBE QUERY NOTIFICATIONS TO [servicebroker];
GRANT CONTROL ON SCHEMA::[dbo] TO [servicebroker];
GRANT CREATE MESSAGE TYPE TO [servicebroker];

However, I was not getting notifications. Once I gave the sysadmin role, I
got the notifications. Am I missing something here?
Are these permissions not sufficient (without sysdamin)?

Thanks
Swanand
